packages/nuxt/src/head/runtime/plugins/unhead.ts

```diff
@@ -11,17 +12,16 @@ export default defineNuxtPlugin({
     if (import.meta.client) {
       // pause dom updates until page is ready and between page transitions
       let pauseDOMUpdates = true
-      const unpauseDom = () => {
+      const syncHead = async () => {
         pauseDOMUpdates = false
-        // trigger the debounced DOM update
-        head.hooks.callHook('entries:updated', head)
+        await renderDOMHead(head)
       }
       head.hooks.hook('dom:beforeRender', (context) => { context.shouldRender = !pauseDOMUpdates })
       nuxtApp.hooks.hook('page:start', () => { pauseDOMUpdates = true })
       // wait for new page before unpausing dom updates (triggered after suspense resolved)
-      nuxtApp.hooks.hook('page:finish', unpauseDom)
+      nuxtApp.hooks.hook('page:finish', syncHead)
```

Comment:
Ideally we'd do this but then it doesn't solve the issues linked 🤔

There's an experimental flag we can enable that should make the initial SSR hydration 0ms to overcome this overhead.

The risk I see outside of that is if someone has a promise on one of the head tags that is extra slow to resolve it would slow the page transition. Although maybe that's useful for the title (as outlined in the issue)